src/Symfony/Component/Cache/DependencyInjection/CachePoolPass.php

```diff
@@ -76,7 +76,7 @@ public function process(ContainerBuilder $container)
             }
             $name = $tags[0]['name'] ?? $id;
             if (!isset($tags[0]['namespace'])) {
-                $tags[0]['namespace'] = $this->getNamespace($seed, $name);
+                $tags[0]['namespace'] = $this->getNamespace($seed, $name.$adapter->getClass());
```

Comment:
we should take the class from the first `ChildDefinition` that defines one:
`$class = $adapter->getClass()` before the loop
`$class = $class ?: $adapter->getClass()` inside the loop